File: ver_cpp/T04.cpp
// https://leetcode.cn/problems/valid-perfect-square/
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    bool isPerfectSquare(int num)
    {
        int ans = 0;
        for (int i = 0; i <= num; ++i)
        {
            if ((long long)i * i <= num)
            {
                ans = i;
            }
            else
            {
                break;
            }
        }
        return (long long)ans * ans == num;
    }
};

int main()
{
    int num;
    Solution s;
    cin >> num;
    cout << s.isPerfectSquare(num) << endl;
    return 0;
}

File: ver_py/T04.py
# https://leetcode.cn/problems/valid-perfect-square/
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        ans = 0
        for i in range(1, num+1):
            if i*i <= num:
                ans = i
            else:
                break
        return ans*ans == num
    
if __name__ == "__main__":
    num = int(input())
    print(Solution().isPerfectSquare(num))